src/main/java/graphql/annotations/GraphQLAnnotations.java

```diff
@@ -588,12 +588,21 @@ private GraphQLOutputType internalGetGraphQLConnection(AccessibleObject field, G
         assert wrappedType instanceof GraphQLObjectType;
```

Comment:
I know it is not relevant for this PR, but why does this assert make sense? we allow other types (interface,enum,union etc.)

Reply:
Yes, completely agree, the assert does not make sense. The wrapped type is an OutputType ( actually, a type from TYPES_FOR_CONNECTION ). I tried to simplify the code by being more specific on the types. I also moved out the operation on the builder, as I don't think a `getGraphQLConnection()` method should mutate something, in my opinion it's better to keep it outside.